server/controllers/notification.js

```diff
@@ -0,0 +1,176 @@
+const Notification = require("../models/Notification");
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+const Profile = require("../models/Profile");
+
+exports.createNotification = asyncHandler(async (req, res, next) => {
+  const {
+    user,
+    body: { notifyType, title, description, receivers },
+  } = req;
+
+  const sender = await Profile.findOne({ userId: user.id });
+
+  if (!sender) {
+    res.status(401);
+    throw new Error("Not authorized");
+  }
```

Comment:
Alternatively, we can assume that the `protect` middleware won't forward any requests from unauthenticated users (i.e. if `jwt` is valid, that should mean user exists and authenticated).

Reply:
While, yes, we can assume that. What the protect middleware does not check when authorizing a user is our database. If someone has a deleted account but a valid jwt token, then they can still access this route.

Reply:
@KookiKodes In that case, why don't we check for a valid/profile user within our `protect` middleware.